### ISSUE TITLE
Dbus: Getting empty reply messages when somebody "blocks and leaves" group

### ISSUE THREAD

**Issue by bublath** [AsamK/signal-cli#516]
My Dbus user (registered land line) owns a couple of groups.
When another user, that is joined to one of those groups, chooses to "block and leave" the group, I am getting a message of the type "MessageReceived" for every group (probably for every group the leaving contact was member as well) with an empty message.
Note: this only happens with "block and leave" - just doing a "leave" does not trigger anything.

Shouldn't something like this come via a SyncMessageReceived?
I actually never received any message of that type so far.

Refers to signal-cli 0.7.4.

**Comment by AsamK** [comment]
SyncMessageReceived is only sent by linked devices, e.g. Signal-Desktop.

When a user blocks a group the signal clients creates a new profile key and updates all his remaining groups with the new key. Those group update messages are received as empty messages over dbus.
The profile key is renewed, so members of the old group don't have access to the user's profile anymore and can't readd him to the group.

**Comment by bublath** [comment]
I could actually re-invite the user (message: xxx added you to the group) but the app still asked for Delete or Unblock. Since the user did not receive messages, that probably is correct behaviour.
However strange: When trying to delete the group right after "unblock and leave", I got an error.
After I got "re-added" the delete actually worked.

Probably that is rather an issue of the app than signal-cli.
Also my test user keeps accumulating "dead" groups since the "delete" feature is not available in Dbus (when I block a group, the getGroupIDs still lists the group and I also cannot query the "blocked" status for group or users and would need to track them on my own.)
